course_catalog/etl/youtube.py

```diff
@@ -101,3 +102,41 @@ def extract():
             yield from get_videos_for_channel(
                 youtube_client, channel["channel_id"], channel["offered_by"]
             )
+
+
+def transform_single_video(offered_by, raw_video_data):
+    """
+    Transforms raw video data into normalized data structure for single video
+
+    Args:
+        offered_by (string)
+        raw_video_data (dict)
+
+    Returns:
+        dict: normalized video data
+    """
+    return {
+        "video_id": raw_video_data["id"],
+        "platform": offered_by,
+        "full_description": raw_video_data["snippet"]["description"],
+        "image_src": raw_video_data["snippet"]["thumbnails"]["high"]["url"],
+        "last_modified": raw_video_data["snippet"]["publishedAt"],
+        "published": True,
+        "url": "https://www.youtube.com/watch?v=" + raw_video_data["id"],
+        "raw_data": raw_video_data,
+    }
+
+
+@log_exceptions("Error transforming youtube data", exc_return_value=[])
+def transform(raw_videos_data):
+    """
+    Transforms raw video data into normalized data structure
+
+    Args:
+        iterable that yields raw video data in the the form of (offered_by, data) tuples
```

Comment:
The arg syntax format should have the variable name and type like this:
```suggestion
        raw_videos_data (iterable of tuples) tuple has the structure (offered_by, data)
```